doc/user-manual/language/cubical.lagda.rst

```diff
@@ -97,10 +113,576 @@ of the interval ``i0`` and ``i1`` are the bottom and top elements, respectively
+transport operation is generalized in the sense that it lets us
+specify where it is the identity function
+
+.. code-block:: agda
+
+  transp : ∀ {ℓ} (A : I → Set ℓ) (r : I) (a : A i0) → A i1
+
+When calling ``transp A r a`` Agda makes sure that ``A`` is constant
+on ``r`` so that ``transp A i1 a`` is definitionally ``a``. This lets
+us define regular transport as
+
+.. code-block:: agda
+
+  transport : ∀ {ℓ} {A B : Set ℓ} → A ≡ B → A → B
+  transport p a = transp (λ i → p i) i0 a
+
+By combining the transport and min operations we can define the
+induction principle for paths:
+
+.. code-block:: agda
+
+  J : ∀ {ℓ} {A : Set ℓ} {x : A} (P : ∀ y → x ≡ y → Set ℓ)
+        (d : P x refl) {y : A} (p : x ≡ y)
+      → P y p
+  J P d p = transport (λ i → P (p i) (λ j → p (i ∧ j))) d
+
+One subtle difference between paths and the propositional equality
+type of Agda is that the computation rule for ``J`` does not hold
+definitionally. If ``J`` is defined using pattern-matching as in the
+standard library then this holds, however as the path types are not
+inductively defined this does not hold for the above definition of
+``J``. In particular, transport in a constant family is only the
+identity function up to a path which implies that the computation rule
+for ``J`` only holds up to a path:
+
+.. code-block:: agda
+
+  transportRefl : ∀ {ℓ} {A : Set ℓ} (x : A) → transport refl x ≡ x
+  transportRefl {A = A} x i = transp (λ _ → A) i x
+
+  JRefl : ∀ {ℓ} {A : Set ℓ} {x : A} (P : ∀ y → x ≡ y → Set ℓ)
+           (d : P x refl) → J P d refl ≡ d
+  JRefl P d = transportRefl d
+
+Internally in Agda the ``transp`` operation computes by cases on the
+type, so for example for Sigma types it is computed elementwise. For
+path types it is however not yet possible to provide the computation
+rule as we need some way to remember the endpoints of the path after
+transporting it. Furthermore, this must work for arbitrary higher
+dimensional cubes (as we can iterate the path types). For this we
+introduce the "homogeneous composition operations" (``hcomp``) that
+generalize binary composition of paths to n-ary composition of higher
+dimensional cubes.
+
+
+Partial elements and systems
+----------------------------
+
+In order to describe the homogeneous composition operations we need to
+be able to write partially specified n-dimensional cubes (i.e. cubes
+where some faces are missing). Given an element of the interval ``r :
+I`` there is a predicate ``IsOne`` which represents the constraint ``r
+= i1``. This comes with a proof that ``i1`` is in fact equal to ``i1``
+called ``1=1 : IsOne i1``. We use Greek letters like ``φ`` or ``ψ``
+when such an ``r`` should be thought of as being in the image of
+``IsOne``.
+
+Using this we introduce a type of partial elements called ``Partial φ
+A``, this is a special version of ``IsOne φ → A`` with a more
+extensional judgmental equality. The idea is that ``Partial φ A`` is
```

Comment:
@Saizan I think I got the remark about "a more extensional equality" from you. Can you please remind me of what it meant more exactly?

Reply:
Much like in cubicaltt for systems, two elements ``u, t : Partial φ A`` are equal if for every face map `α` that `φ` decomposes into, we have `uα = tα`.

In the docs one might want to be less technical though.

Reply:
Thanks, I clarified this by saying that u and t are considered equal if they specify the same subcube (so faces can be given in different order, etc.)